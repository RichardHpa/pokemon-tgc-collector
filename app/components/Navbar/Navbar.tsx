import { Link, Form } from "@remix-run/react";
import { Button } from "~/components/Button";
import { ThemeToggle } from "~/components/ThemeToggle";

import { useOptionalUser } from "~/utils";

export const Navbar = () => {
  const user = useOptionalUser();

  return (
    <nav className="border-gray-200 px-2 py-2.5 sm:px-4">
      <div className="container mx-auto flex flex-wrap items-center justify-between border-b py-2">
        <div className="flex items-center space-x-2">
          <Link
            className="flex p-0 text-2xl tracking-tighter text-gray-700 no-underline dark:text-white"
            to="/"
          >
            <span className="inline-block font-medium">
              Pokemon Card Checker
            </span>
          </Link>

          {/* We are just keeping this in here until we edit the cypress tests */}
          <Button as={Link} to="/notes" variant="text" color="neutral">
            notes
          </Button>

          <Button as={Link} to="/expansions" variant="text" color="neutral">
            Expansions
          </Button>
        </div>
        <div className="flex items-center space-x-2">
          {user ? (
            <>
              <p className="font-medium text-gray-700">{user.email}</p>
              <Form action="/logout" method="post">
                <Button variant="text" color="neutral" type="submit">
                  Logout
                </Button>
              </Form>
            </>
          ) : (
            <>
              <Button as={Link} to="/join" variant="text" color="neutral">
                Sign up
              </Button>
              <Button as={Link} to="/login" variant="text" color="neutral">
                Log In
              </Button>
            </>
          )}
          <ThemeToggle />
        </div>
      </div>
    </nav>
  );
};
