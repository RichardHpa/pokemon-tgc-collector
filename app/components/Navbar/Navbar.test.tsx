import { render } from "test/helpers/render";
import { screen } from "@testing-library/react";
import { Navbar } from "./Navbar";

vi.mock("~/utils", () => {
  return {
    useOptionalUser: vi.fn(),
  };
});

describe("Navbar", () => {
  afterEach(() => {
    vi.clearAllMocks();
  });
  test("renders Navbar", () => {
    render(<Navbar />);

    const link = screen.getByRole("link", { name: /Pokemon Card Checker/i });
    expect(link).toBeInTheDocument();
  });

  test("renders all links", () => {
    render(<Navbar />);
    expect(screen.getAllByRole("link")).toHaveLength(4);

    const setsLink = screen.getByRole("link", { name: /sets/i });
    expect(setsLink).toBeInTheDocument();

    const loginLink = screen.getByRole("link", { name: /login/i });
    expect(loginLink).toBeInTheDocument();

    const registerLink = screen.getByRole("link", { name: /sign up/i });
    expect(registerLink).toBeInTheDocument();
  });

  test("renders Navbar snapshot", () => {
    const { container } = render(<Navbar />);
    expect(container).toMatchSnapshot();
  });
});
