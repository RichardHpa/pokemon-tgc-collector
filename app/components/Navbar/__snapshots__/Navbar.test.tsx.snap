// Vitest Snapshot v1

exports[`Navbar > renders Navbar snapshot 1`] = `
<div>
  <nav
    class="border-gray-200 px-2 py-2.5 sm:px-4"
  >
    <div
      class="container mx-auto flex flex-wrap items-center justify-between border-b py-2"
    >
      <div
        class="flex items-center space-x-2"
      >
        <a
          class="flex p-0 text-2xl tracking-tighter text-gray-700 no-underline dark:text-white"
          href="/"
        >
          <span
            class="inline-block font-medium"
          >
            Pokemon Card Checker
          </span>
        </a>
        <a
          class="inline-flex px-4 py-2 rounded justify-center items-center font-medium hover:underline text-gray-700 dark:text-white"
          href="/notes"
        >
          notes
        </a>
        <a
          class="inline-flex px-4 py-2 rounded justify-center items-center font-medium hover:underline text-gray-700 dark:text-white"
          href="/expansions"
        >
          Expansions
        </a>
      </div>
      <div
        class="flex items-center space-x-2"
      >
        <a
          class="inline-flex px-4 py-2 rounded justify-center items-center font-medium hover:underline text-gray-700 dark:text-white"
          href="/join"
        >
          Sign up
        </a>
        <a
          class="inline-flex px-4 py-2 rounded justify-center items-center font-medium hover:underline text-gray-700 dark:text-white"
          href="/login"
        >
          Log In
        </a>
        <button
          class="inline-flex align-middle justify-center rounded-full p-2 text-gray-500 hover:bg-gray-50 border-gray-500 dark:text-gray-500 border !p-[calc(0.5rem-1px)]"
        >
          <svg
            aria-hidden="true"
            class="h-6 w-6 inline-block"
            fill="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 2.25a.75.75 0 01.75.75v2.25a.75.75 0 01-1.5 0V3a.75.75 0 01.75-.75zM7.5 12a4.5 4.5 0 119 0 4.5 4.5 0 01-9 0zM18.894 6.166a.75.75 0 00-1.06-1.06l-1.591 1.59a.75.75 0 101.06 1.061l1.591-1.59zM21.75 12a.75.75 0 01-.75.75h-2.25a.75.75 0 010-1.5H21a.75.75 0 01.75.75zM17.834 18.894a.75.75 0 001.06-1.06l-1.59-1.591a.75.75 0 10-1.061 1.06l1.59 1.591zM12 18a.75.75 0 01.75.75V21a.75.75 0 01-1.5 0v-2.25A.75.75 0 0112 18zM7.758 17.303a.75.75 0 00-1.061-1.06l-1.591 1.59a.75.75 0 001.06 1.061l1.591-1.59zM6 12a.75.75 0 01-.75.75H3a.75.75 0 010-1.5h2.25A.75.75 0 016 12zM6.697 7.757a.75.75 0 001.06-1.06l-1.59-1.591a.75.75 0 00-1.061 1.06l1.59 1.591z"
            />
          </svg>
        </button>
      </div>
    </div>
  </nav>
</div>
`;
